ddtrace/tracer/span.go

```diff
@@ -92,51 +92,74 @@ func (s *span) SetTag(key string, value interface{}) ddtrace.Span {
 	if s.finished {
 		return s
 	}
+	if key == ext.Error {
+		return s.setTagError(value)
+	}
+	if v, ok := value.(string); ok {
+		return s.setTagString(key, v)
+	}
 	if v, ok := toFloat64(value); ok {
-		// sent as numeric value, so we can store it as a metric
-		switch key {
-		case ext.SamplingPriority:
-			// setting sampling priority per spec
-			s.Metrics[samplingPriorityKey] = v
-		default:
-			s.Metrics[key] = v
+		return s.setTagNumeric(key, v)
+	}
+	// not numeric, not a string and not an error, the likelihood of this
+	// happening is close to zero, but we should nevertheless account for it.
```

Comment:
Speaking of which, I'm not sure we actually test for this within the unit test. We have one test with `false` https://github.com/DataDog/dd-trace-go/blob/a3458eb028f686b5196670c744e68705eaea6cbb/ddtrace/tracer/span_test.go#L145 but that's pretty much it AFAIK. It's a nitpick anyway, and as you mention, this is possibly a corner case.

Reply:
Actually you're right, we should have a test for it. The line you mention does not reach this point, it goes into the error branch further up in the method. I've added the test. Thanks for looking.